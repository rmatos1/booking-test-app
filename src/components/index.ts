export * from "./bookingDescription";
export * from "./bookingForm";
export * from "./images";
export * from "./inputGroup";
export * from "./navBar";
export * from "./styledComponents";
export * from "./testComponent";
