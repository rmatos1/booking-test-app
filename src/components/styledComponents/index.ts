export * from './buttons';
export * from './formComponents';
export * from './layouts';
export * from './lists';
export * from './modalComponents';
export * from './texts';
