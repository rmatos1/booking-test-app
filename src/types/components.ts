import { ReactNode } from "react";

export interface IComponentWithChildren {
  children?: ReactNode;
}

export interface IBaseDrawer extends IComponentWithChildren {
  isDrawerVisible: boolean;
  onCloseDrawer: () => void;
}
