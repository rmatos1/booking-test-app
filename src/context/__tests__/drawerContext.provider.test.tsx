import { describe, test } from '@jest/globals';
import { fireEvent, render } from '@testing-library/react';

import { ReactNode } from 'react';
import { DrawerContext, IDrawerContext } from '..';
import { TestComponent } from '../../components';

const testElement: ReactNode = <div data-testid="test-element" />;

const testConsumerContext = (
  children: (value: IDrawerContext) => ReactNode
) => {
  return <DrawerContext.Consumer>{children}</DrawerContext.Consumer>;
};

const setup = (children: ReactNode): JSX.Element => {
  return <TestComponent>{children}</TestComponent>;
};

describe('<DrawerProvider />', () => {
  test('should render correctly', () => {
    const wrapper = render(setup(testElement));

    expect(wrapper).toMatchSnapshot();
  });

  test('should render the defined children element', () => {
    const wrapper = render(setup(testElement));

    const children = wrapper.getByTestId('test-element');

    expect(children).toBeDefined();
  });

  test('should render the specified element after the click', () => {
    const context = testConsumerContext((value) => (
      <>
        <button
          onClick={() => value.setIsDrawerVisible(true)}
          data-testid="booking-button"
        ></button>

        {value.isDrawerVisible && <div data-testid="test-element" />}
      </>
    ));

    const wrapper = render(setup(context));

    const bookingButton = wrapper.getByTestId('booking-button');
    fireEvent.click(bookingButton);

    const testElement = wrapper.getByTestId('test-element');

    expect(testElement).toBeDefined();
  });
});
