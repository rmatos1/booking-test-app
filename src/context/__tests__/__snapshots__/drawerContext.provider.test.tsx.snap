// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerProvider /> should render the defined children element correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="test-element"
      />
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="test-element"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<DrawerProvider /> should render the specified element after the click 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        data-testid="booking-button"
      />
      <div
        data-testid="test-element"
      />
    </div>
  </body>,
  "container": <div>
    <button
      data-testid="booking-button"
    />
    <div
      data-testid="test-element"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
